libraries/AP_Winch/AP_Winch.cpp

```diff
@@ -0,0 +1,123 @@
+#include "AP_Winch.h"
+#include "AP_Winch_Servo.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_Winch::var_info[] = {
+    // @Param: ENABLE
+    // @DisplayName: Winch enable/disable
+    // @Description: Winch enable/disable
+    // @User: Standard
+    // @Values: 0:Disabled, 1:Enabled
+    AP_GROUPINFO_FLAGS("_ENABLE", 0, AP_Winch, _enabled, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: TYPE
+    // @DisplayName: Winch Type
+    // @Description: Winch enable/disable
+    // @User: Standard
+    // @Values: 0:None,1:Servo with encoder
```

Comment:
That's a good point about not needing the 0:None option and setting the default to "1".  I've done this in a follow-up commit.  I somehow feel we should reserve the type of zero to mean none but I don't have a very specific reason.

Reply:
Yes, I can see that None would be useful in case of multiple winch instances, so I'm OK with leaving that reserved for the future. I just think that with the single instance there is in this PR, it isn't worth it having the user have to set ENABLE and TYPE.

Reply:
Yes, I agree it's slightly annoying to have both enable and type.  I did think of combining them but I also want to hide the parameters because 99.9% of users will never use a winch.  I thought hiding the other parameters with the winch-type parameter might be odd.. at least people might not be expecting a parameter ending in "_TYPE" to have the same parameter hiding effect as we sometimes do with "_ENABLE" parameters.